src/utils/updateDBIcons.ts

```diff
@@ -0,0 +1,129 @@
+    if (!notifiedByHook) {
+      // if not notified by hook, we need to make sure that there's a new release first
+      const eosIconsTAGS = await axios.get(configs.GITLAB_TAGS_API)
+      const latestReleaseDate = eosIconsTAGS.data[0].commit.committed_date
+      if (latestReleaseDate <= currVersionDate) {
+        updateIconsLogger.logInfo('', { message: 'icons is up to date, updating process is finished' })
+        return
+      }
+    }
+    updateIconsLogger.logInfo('', { message: 'The process of updating the icons is started' })
+    console.time('Updating the icons Process')
+    // Retrieve the icons' JSON file content:
+    const iconsJSONfile = await axios.get(getEncodedLink('dist/js/eos-icons.json'))
+    const gitlabIcons = iconsJSONfile.data as IconInterface[]
+    updateIconsLogger.logInfo('Retriving the icons', { message: 'Finished' })
+    // Update the local eos-icons package and move the files using grunt:
+    await pcmdGet('npm update eos-icons')
+    await pcmdGet('grunt -b ./src moveFiles')
+    updateIconsLogger.logInfo('Updating the local package, and moving the files', { message: 'Finished' })
+
+    /**
+     **********************************************
+     The process of retrieving all the data that will be used in calculations:
+     **********************************************
+     */
+
+    // Current icons:
+    const currentDBIcons = await iconsServices.getAllIcons() as IIconsModel[] | IconInterface[]
+    updateIconsLogger.logInfo('Retrieving the needed data from the databases and APIs', { message: 'Finished' })
+
+    /**
+     **********************************************
+     The preparation of the objects
+     **********************************************
+     */
+    const newIcons: IconInterface[] = []
+    const deletedIcons: string[] = []
+    // for auditing purposes:
+    const namesOfDeletedIcons: string[] = []
+    const namesOfUpdatedIcons: string[] = []
+    const namesOfNewIcons: string[] = []
+    // This will get all the newer icons (new/modified icon):
+    const addedAndUpdatedIcons = gitlabIcons.filter(icon => isNewIcon(icon.date, currVersionDate))
+    updateIconsLogger.logInfo('The preparation of the files', { message: 'Finished' })
+
+    // Iterate through the icons:
+    for (const iconDetails of addedAndUpdatedIcons) {
+      // ADD the SVG codes to the icon's object:
+      const filledSVG = await getFilled(iconDetails.name)
+      if (filledSVG === 'NOT FOUND') {
+        continue
+      }
+      iconDetails.svg = filledSVG
+      // IF icon has an outlined version, get the svg of it:
+      if (iconDetails.hasOutlined) {
+        const outlinedSVG = await getOutlined(iconDetails.name)
+        if (outlinedSVG === 'NOT FOUND') {
+          // if not found, use the same svg code of the filled version
+          iconDetails.svgOutlined = iconDetails.svg
+        } else {
+          iconDetails.svgOutlined = outlinedSVG
```

Comment:
As per our discussions in slack, if an icon should have an outlined version, and we couldn't find the file, we will use the same `svg` code.